datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -747,10 +770,10 @@ macro_rules! get_data_page_statistics {
                 Some(DataType::Boolean) => Ok(Arc::new(
                     BooleanArray::from_iter(
```

Comment:
We could use BooleanBuilder instead? That would likely be more efficient.

Reply:
> We could use BooleanBuilder instead? That would likely be more efficient.

Seems reasonable! Maybe it can work to remove the `collect` in  `BooleanArray` case.